lib/pbench/test/unit/server/test_datasets_compare.py

```diff
@@ -53,144 +54,107 @@ def query_api(
-            self, benchmark_name, input_type, data_stream
-        ) -> JSON:
-            return {"status": "success", "json_data": "quisby_data"}
-
-        monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
-        monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
-        monkeypatch.setattr(
-            QuisbyProcessing, "compare_csv_to_json", mock_compare_csv_to_json
-        )
-
-        response = query_get_as(["uperf_1", "uperf_2"], "test", HTTPStatus.OK)
-        assert response.json["status"] == "success"
-        assert response.json["json_data"] == "quisby_data"
+    def test_unsuccessful_get_with_incorrect_data(self, query_get_as, monkeypatch):
+        def mock_extract(_tarball_path: Path, _path: str) -> str:
+            raise "IncorrectData"
 
-    def test_unauthorized_user_compares_public_datasets(
-        self, query_get_as, monkeypatch
-    ):
         def mock_compare_csv_to_json(
             self, benchmark_name, input_type, data_stream
         ) -> JSON:
-            return {"status": "success", "json_data": "quisby_data"}
+            return {"status": "failed", "exception": "Unsupported Media Type"}
 
         monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(self.MockTarball, "extract", mock_extract)
         monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
         monkeypatch.setattr(
             QuisbyProcessing, "compare_csv_to_json", mock_compare_csv_to_json
         )
+        query_get_as(["uperf_1", "uperf_2"], "test", HTTPStatus.INTERNAL_SERVER_ERROR)
 
-        response = query_get_as(["fio_1", "fio_2"], None, HTTPStatus.OK)
-        assert response.json["status"] == "success"
-        assert response.json["json_data"] == "quisby_data"
-
-    def test_unsuccessful_get_with_incorrect_data(self, query_get_as, monkeypatch):
-        def mock_find_dataset_with_incorrect_data(self, dataset):
-            class MockTarball(object):
-                tarball_path = Path("/dataset/tarball.tar.xz")
-
-                def extract(tarball_path, path) -> str:
-                    return "IncorrectData"
-
-            return MockTarball
-
-        def mock_compare_csv_to_json(
-            self, benchmark_name, input_type, data_stream
-        ) -> JSON:
-            return {"status": "failed", "exception": "Unsupported Media Type"}
+    def test_tarball_unpack_exception(self, query_get_as, monkeypatch):
+        def mock_extract(_tarball_path: Path, _path: str):
+            raise TarballUnpackError
 
         monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(self.MockTarball, "extract", mock_extract)
         monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
-        monkeypatch.setattr(
-            QuisbyProcessing, "compare_csv_to_json", mock_compare_csv_to_json
-        )
         query_get_as(["uperf_1", "uperf_2"], "test", HTTPStatus.INTERNAL_SERVER_ERROR)
```

Comment:
Based on looking at the coverage, I don't think this test is working correctly.  I think we're expecting this test to raise an `APIInternalError` at [line 110](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/job/pbench-ci-multi/job/PR-3470/11/cobertura/server_api_resources/datasets_compare_py/#110), but Cobertura claims that that line is never executed.  Presumably, the exception is being raised at line 102, instead, but I don't know why.  (Is there some reason why user `test` doesn't have access to datasets `uperf_1` and `uperf_2` such that the `mock_find_dataset()` call is raising an exception?)

Reply:
Yeah @webbnh, It was due to this `raise TarballUnpackError`  as you mentioned in the earlier comment, it expects the instance of the `TarballUnpackError` which I missed. After adding this suggestion 
`raise TarballUnpackError(_tarball_path, f"Testing unpack exception for path {_path}")` , It was working as expected :)